mono/metadata/threads.c

```diff
@@ -4598,7 +4598,7 @@ mono_thread_interruption_checkpoint_request (gboolean bypass_abort_protection)
 		return NULL;
 	if (!mono_thread_get_interruption_requested (thread))
 		return NULL;
-	if (!bypass_abort_protection && is_running_protected_wrapper ())
+	if (!bypass_abort_protection && !mono_thread_current ()->pending_exception && is_running_protected_wrapper ())
```

Comment:
@vargaz Wasn't the previously set pending exception getting thrown from the managed to native wrapper ?

Reply:
Yes, but is_running_protected_wrapper () does a full stack walk, so it won't throw if there is a remoting wrapper above, hence this change.